openedx/core/lib/gating/services.py

```diff
@@ -0,0 +1,47 @@
+"""
+A wrapper class around requested methods exposed in api.py
+"""
+import types
+
+from . import api as gating_api
+
+
+class GatingService(object):  # pylint: disable=too-few-public-methods
+    """
+    An xBlock service for xBlocks to talk to the Gating api.
+
+    NOTE: This is a Singleton class. We should only have one instance of it!
+    """
+
+    _instance = None
+
+    REQUESTED_FUNCTIONS = [
+        'get_gating_milestone_meta_info',
+        'get_gating_milestone',
+        'is_prereq_met'
+    ]
+
+    def __new__(cls, *args, **kwargs):
+        """
+        This is the class factory to make sure this is a Singleton
+        """
+        if not cls._instance:
+            cls._instance = super(GatingService, cls).__new__(cls, *args, **kwargs)
+        return cls._instance
```

Comment:
Why enforce that this is a Singleton?

Reply:
Wasn't sure about this, I followed the pattern used by Milestones Service which did this. Based on below comments I agree with what you are suggesting.